lib/smack/SmackRep.cpp

```diff
@@ -820,7 +820,11 @@ const Expr* SmackRep::cast(const llvm::ConstantExpr* CE) {
 }
 
 const Expr* SmackRep::cast(unsigned opcode, const llvm::Value* v, const llvm::Type* t) {
-  return Expr::fn(opName(Naming::INSTRUCTION_TABLE.at(opcode), {v->getType(), t}), expr(v));
+  std::string fn = Naming::INSTRUCTION_TABLE.at(opcode);
+  if (fn[1] == 'f' || fn[3] == '2') { //$fptrunc, $fpext, $fp2ui, $fp2si, $ui2fp, $si2fp
```

Comment:
It would be better to list all of the opcodes here, instead of relying on string names.